mlflow/pipelines/steps/evaluate.py

```diff
@@ -120,6 +120,20 @@ def _check_validation_criteria(self, metrics, validation_criteria):
         return summary
 
     def _run(self, output_directory):
+        def warn(*args, **kwargs):
+            import sys
+
+            stacklevel = 1 if "stacklevel" not in kwargs else kwargs["stacklevel"]
+            frame = sys._getframe(stacklevel)
+            filename = frame.f_code.co_filename
+            lineno = frame.f_lineno
+            open(os.path.join(output_directory, "warning_logs.txt"), "a").write(
```

Comment:
What happens if something goes wrong with a stack overflow? Would this log be every displayed to user? Can we test that user experience?

Reply:
I'm not quite following this issue. What would be the problem in the case of stack overflow?

BTW, I have modeled this implementation from the python warnings library https://github.com/python/cpython/blob/main/Lib/warnings.py#L287-L310

Reply:
If the step fails, we show failure card instead of the regular card. My question was, does the warning tab also show on the failure card?